main.go

```diff
@@ -97,9 +97,6 @@ func main() {
 		contentType := ""
 		for firstTime := true; firstTime || bytesRead == len(fileBytes); {
 		    bytesRead, err = file.Read(fileBytes)
-		    if err != nil {
-			fmt.Printf("%s\n", err);
-		    }
```

Comment:
On Tue, Feb 16, 2016 at 9:10 AM, Martin Atkins <notifications@github.com>
wrote:

> In main.go
> <https://github.com/saymedia/terraform-s3-dir/pull/3#discussion_r53042654>
> :
>
> > @@ -97,9 +97,6 @@ func main() {
> >  		contentType := ""
> >  		for firstTime := true; firstTime || bytesRead == len(fileBytes); {
> >  		    bytesRead, err = file.Read(fileBytes)
> > -		    if err != nil {
> > -			fmt.Printf("%s\n", err);
> > -		    }
>
> I didn't catch this on the first pass but it looks like this wasn't
> actually trapping the error anyway, so later on when we write fileBytes
> into the hasher we're presumably writing either an empty buffer or whatever
> happened to be left in the buffer from a previous read.
>
I was having a bit of a hard time debugging this.  $len in gdb didn't work
as I think the documentation was telling me it should.  According to the
docs file.Read should return 0, io.EOF on the error I saw which should fall
though correctly assuming the array was really of length 0.


> It looks like the existing pattern elsewhere in this thing was to print a
> message to stderr and return nil, which I guess causes it to skip over
> the files it has trouble with. (I don't really remember exactly what went
> into that now, honestly...) So perhaps the appropriate thing here is to do
> similarly, skipping over this particular file with a warning and then
> continuing with the others.
>

Will do.

        Michael


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saymedia/terraform-s3-dir/pull/3/files#r53042654>.
>